source/_integrations/heos.markdown

```diff
@@ -69,73 +67,65 @@ Use the sign-out service to sign the connected controller out of a HEOS account.
 
 You can play a HEOS favorite by number or name with the `media_player.play_media` service. Example service data payload:
 
-```json
-{
-  "entity_id": "media_player.office",
-  "media_content_type": "favorite",
-  "media_content_id": "1"
-}
+```yaml
+entity_id: media_player.office
+media_content_type: "favorite"
+media_content_id: "1"
 ```
 
-| Attribute              | Description
-| ---------------------- | ---------------------------------------------------------|
-| `entity_id`            | `entity_id` of the player
-| `media_content_type`   | Set to the value `favorite`
-| `media_content_id`     | The nubmer (i.e., `1`) or name (i.e., `Thumbprint Radio`) of the HEOS favorite
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| `entity_id`            | no       | `entity_id` of the player
```

Comment:
`entity_id` is optional, because alternatively an area could be passed. This applies to all others as well.

Reply:
I cannot find any reference to this in any of the player docs. The official docs also states that `entity_id` is required: https://www.home-assistant.io/integrations/media_player/#service-media_playerplay_media
Are the official `media_player` docs wrong as well?

Reply:
That would be incorrect as well in that case.

Let me guide you through the codebase:

The entity services schemas are created in the service register call, using the `make_entity_service_schema` method.

For example, the `play_media` is defined here: <https://github.com/home-assistant/core/blob/dev/homeassistant/components/media_player/__init__.py#L307>

This is the `make_entity_service_schema` function:

<https://github.com/home-assistant/core/blob/62f2ee5f60586402e90ac868354550571e96cb2a/homeassistant/helpers/config_validation.py#L790>

As you can see in that function, entity ID is declared optional:


`vol.Optional(ATTR_ENTITY_ID): comp_entity_ids,`

<https://github.com/home-assistant/core/blob/62f2ee5f60586402e90ac868354550571e96cb2a/homeassistant/helpers/config_validation.py#L798>

Conclusion: `entity_id` is optional.

Reply:
Okay, I see. I have updated the docs and tried to explain that either `entity_id` or `area_id` is required.